### PR TITLE
Add missing nil check in register_deadline

### PR DESCRIPTION
**Pull request by byucesoy**
Previously we didn't add nil check before parsing deadline_at, because we had check on the deadline_target already. However for the deadlines we specify for Prog exit, we put nil target. In such cases we need to nil check on deadline_at as well. In the current logic, register_deadline only registers a deadline if it satisfies any of the 3 conditions;
- There is no deadline set (determined by checking if deadline_at is nil)
- Current deadline is set for a further time than new deadline registration
- Current deadline is set for another target

Apart from that, we also start to clear pages if it exist and the target is changed.

### CODE REVIEW

File: prog/base.rb [hunk 1]
```diff
@@ -183,12 +183,20 @@ def hop(label)
   end
 
   def register_deadline(deadline_target, deadline_in)
-    return if strand.stack.first["deadline_target"] == deadline_target && Time.parse(strand.stack.first["deadline_at"].to_s) < Time.now + deadline_in
+    if strand.stack.first["deadline_at"].nil? ||
```
Comment (fdr) [inline prog/base.rb:186]: Seems like you can clean this up with '(binding = expr)' 

Such forms are frequently verboten in C code bases, but I find them quite useful in Ruby.

Comment (byucesoy) [reply prog/base.rb]: It got cleaned up a bit, but not much. I also put strand.stack.first to a variable to clean up a bit more. Though I'm not very happy with the end result. Do you see any other clean up opportunities?

**Review by fdr — APPROVED** [review]
Without much conviction, I did a couple of variable bindings in conditional arms to avoid digging around in the hash all the time. You could opt to squash it, or not.

**Comment by byucesoy** [comment]
Merging with your changes as well.

**Pull request merged**